kafka/handle.go

```diff
@@ -205,3 +246,45 @@ func (h *handle) cgoGet(cgoid int) (cg cgoif, found bool) {
 
 	return cg, found
 }
+
+func (h *handle) setOAuthBearerToken(oauthBearerToken OAuthBearerToken) error {
+	cTokenValue := C.CString(oauthBearerToken.TokenValue)
+	defer C.free(unsafe.Pointer(cTokenValue))
+	cPrincipal := C.CString(oauthBearerToken.Principal)
+	defer C.free(unsafe.Pointer(cPrincipal))
+	cErrstrSize := C.size_t(512)
+	cErrstr := (*C.char)(C.malloc(cErrstrSize))
+	defer C.free(unsafe.Pointer(cErrstr))
+	cExtensions := make([]*C.char, 2*len(oauthBearerToken.Extensions))
+	extensionSize := 0
+	for key, value := range oauthBearerToken.Extensions {
+		cExtensions[extensionSize] = C.CString(key)
+		defer C.free(unsafe.Pointer(cExtensions[extensionSize]))
+		extensionSize++
+		cExtensions[extensionSize] = C.CString(value)
+		defer C.free(unsafe.Pointer(cExtensions[extensionSize]))
+		extensionSize++
+	}
+	var cExtensionsToUse **C.char
+	if extensionSize == 0 {
```

Comment:
`cExtensionsToUse` is initialized to nil so this can be simplified.
```suggestion
	if extensionSize > 0 {
		cExtensionsToUse = (**C.char)(unsafe.Pointer(&cExtensions[0]))
	}
```

Reply:
👍